### DISCUSSION TITLE
Heroku One-Click Deploy

### DISCUSSION THREAD

**Discussion by sammachin** [node-red/designs#58] [Feature Requests]
I'd like to add a heroku install package to the docs to make it easy for people to get started with Node-RED in the cloud, I've based this off https://github.com/joeartsea/node-red-heroku but made some tweaks to remove the landing page and also make a few more settings configurable through Heroku Config Vars.
It uses a heroku postgres db for the storage and I've added a new node that will work with that DB for easy data storage in flows too.

Could we create a new repo in the node-red org to host this install from and then I can also submit a PR for the docs to link to it.

**Comment by dceejay** [comment]
is this new node is a general purpose postgres node ? Or specific to this ? Is there a storage plugin also ? Would it work with context also ? 

(reason I ask is that there are quite a few contrib nodes out there in various states of working/features so it would be good to have a more official one that is maintained as postgres does seem to come up a lot.)

**Reply by sammachin** [reply]
Its got a couple of 'tweaks' the connection details are a single URL either as a string, env var or the DATABASE_URL env var that heroku uses. Would be easy enough to add in the option to specify each value separately too like the other nodes.

The other thing is that heroku uses self-signed SSL certs on their postgres instances, so this node as `ssl: {rejectUnauthorized: false}` set on the connection, but it would be simple enough to make that a configurable option, I don't think false would cause issues with other PG hosts that use signed certs other than it reduces the security a little? But making it a checkbox in the config and defaulting to false for heroku would seem logical, its not an easy errror to debug otherwise (took me a while)

Other than that I just cleaned up the  README and Help and added some example flows.
The code is at https://github.com/sammachin/node-red-contrib-herokupg

**Reply by dceejay** [reply]
Aha - by naming it herokupg then it is fairly obvious that it what it is for so I have no problem with that if you want to keep it "single use" - but likewise happy if you wanted to make it more general also (eg node-red-node-postgres) if Nick agreed.

**Reply by sammachin** [reply]
I'll have a go at making it more "multi-purpose" in a branch and see what that looks like

**Comment by sammachin** [comment]
Regarding the context storage the plugin doesn't currently do context but I think it would be a good addition especially as heroku will kill off apps that have been idle for 30 mins so you could find yourself loosing context data. I'll add an issue to the project for it and take a look

**Reply by dceejay** [reply]
So is this just this discussion about just the node ? Or a storage plugin ? or both ? (and I must admit I'm a bit hazy on whether a storage plugin (for flows) now automatically can be used for context (for nodes) - or if they are different APIs)

**Reply by sammachin** [reply]
Well the original topic was about having a one-click deploy to heroku with the repo on the node-red org, The heroku deploy uses a postgres storage plugin but thats separate from context still I believe.

I'm then proposing to add the new postgres node to the node-red install be default in the same way that the IBM Cloud instances of node-red have the cloudant node installed.

Sorry getting things confused with all the postgres!

**Reply by knolleary** [reply]
This discussion is about the heroku getting started app - for which the herokupg node is one small part of.

@sammachin do you have a link to your repo for the heroku getting started app?

**Reply by sammachin** [reply]
Yes sorry forgot to add that! https://github.com/sammachin/node-red-heroku

**Reply by dceejay** [reply]
Ah right - so yes it's a dedicated node really. And a storage plugin for the flows (a la  Cloudant) - all good... carry on as you were.

... but also adding context would be neat ;-)

**Reply by sammachin** [reply]
Yes I just figured that if you were running on Heroku and had a PG database already setup it would be good to steer people towards that for their application storage too
